src/Frontend/Core.hs

```diff
@@ -154,6 +154,22 @@ type GetH = Get '[HTML, PlainText]
 type PostH p = Post '[HTML] (PostResult p)
 type FormHandler p = FormH '[HTML, PlainText] (Frame (FormPageRep p)) (FormPageResult p)
 
+-- | A void type for end-points that respond with 303 and thus never return any values.
+data Redirect
```

Comment:
We could introduce a smart constructor `redirect :: URL -> Redirect` that throws an exception.  (thanks @np)